packages/libs/web-common/src/controllers/LegacyMapRedirectHandler.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import QueryString from 'querystring';
+import { edaServiceUrl } from '../config';
+import { useConfiguredAnalysisClient } from '@veupathdb/eda/lib/core/hooks/client';
+import { createComputation } from '@veupathdb/eda/lib/core/components/computations/Utils';
+import { makeNewAnalysis } from '@veupathdb/eda/lib/core';
+import { RouteComponentProps } from 'react-router';
+import { LegacyRedirectState } from '@veupathdb/eda/lib/map/analysis/MapAnalysis';
+
+// Define constants to create new computations and analyses
+const MEGA_STUDY_ID = 'DS_480c976ef9';
+const MEGA_STUDIES_ENTITY_ID = 'EUPATH_0000605';
+const POPBIO_ID_VARIABLE_ID = 'POPBIO_8000215';
+const DESCRIPTOR_TYPE = 'stringSet';
+const REDIRECT_ANALYSIS_DESCRIPTION =
+  'This map was created from a legacy PopBio map link.';
+const DEFAULT_COMPUTATION = createComputation(
+  'pass',
+  undefined,
+  [],
+  [],
+  undefined,
+  'Unnamed computation'
+);
+
+export function LegacyMapRedirectHandler({
+  history,
+  match,
+  location,
+}: RouteComponentProps) {
+  const analysisClient = useConfiguredAnalysisClient(edaServiceUrl);
+  const queryParams = useMemo(
+    () => QueryString.parse(location.search.slice(1)),
+    [location.search]
+  );
+  const [hasCreatedAnalysis, setHasCreatedAnalysis] = useState(false);
+
+  const handleLegacyMapRedirect = useCallback(
+    async (computation, additionalConfig = {}, legacyMapRedirectState) => {
+      if (hasCreatedAnalysis) return;
+      const { analysisId } = await analysisClient.createAnalysis(
+        makeNewAnalysis(MEGA_STUDY_ID, computation, additionalConfig)
+      );
+      setHasCreatedAnalysis(true);
```

Comment:
I'm thinking this should be set before creating the analysis, so that the state us updated as soon as possible.